server/app.js

```diff
@@ -0,0 +1,250 @@
+import '@babel/polyfill';
+import bodyParser from 'body-parser';
+import ms from 'ms';
+import swaggerUi from 'swagger-ui-express';
+import validateEnvironmentVars from './validator';
+import swaggerConfig from '../docs/swagger';
+import routes from './routes';
+import worker from './jobs/worker';
+import fs from "fs";
+import path from "path";
+
+// Get env variables from .env
+dotenv.config();
+
+// Validate environment variables before anything else happens
+validateEnvironmentVars();
+
+const app = express();
+
+const loopSequentially = async (array, callback) => {
+  const resultArray = [];
+
+  /* eslint-disable-next-line no-restricted-syntax */
+  for (const [index, data] of array.entries()) {
+    /* eslint-disable-next-line no-await-in-loop */
+    const result = await callback(data, index);
+    resultArray.push(result);
+  }
+  return resultArray;
+};
+
+const httpMethods = [
+  'get', 'post', 'patch',
+  'delete', 'put',
+  'options', 'head', 'trace',
+  'connect', 'all', 'copy', 'link',
+  'unlink', 'purge', 'lock', 'unlock',
+  'propfind', 'view',
+];
+
+/**
+ * Handles the addition of the given path to the app's swagger schema
+ * @param {string} path The path to add
+ * @param {string} method The HTTP method,
+ * could also be 'use' or 'all' in which case we default to 'get'
+ * @param {Object} [options] Optional swagger definition that will descriptions of the given path.
+ * @returns {void}
+ */
+const addToAppSchema = (path, method, options) => {
+  const otherOptions = options || {};
+  app.schema.paths[path] = {
+    [httpMethods.includes(method) ? method : 'get']: {
+      responses: {},
+      ...otherOptions,
+    },
+    ...app.schema.paths[path],
+  };
+};
+
+/**
+ * Checks whether the given instance is an instance of express.Router.
+ * @param {*} instance The instance to perform the check against.
+ * @return {boolean} true if instance is an instance of express.Router, otherwise false.
+ */
+const isRouter = instance => Object.getPrototypeOf(instance) === express.Router;
+
+/**
+ * Handles the case when an instance of express.Router is given as the last argument to an app's
+ * method
+ * @param {string} definedPath The path for which this router should handle
+ * @param {express.Router} router The router which was given
+ * @param {Object} [options] Optional swagger definition that will descriptions of the given path.
+ * @return {Promise<*>} A Promise that resolves when the express Router has been handled
+ */
+const handleExpressRouter = (definedPath, router, options) => loopSequentially(router.stack,
+  (layer) => {
+    // If the router was given a router,
+    // get the path it handles from the regex expression as a string,
+    // append it to the definedPath,
+    // then call this function recursively with it as the first argument,
+    // it's handle as the second argument, and the given options as the third
+    if (layer.name === 'router') {
+      const regexPath = layer.regexp.toString().split('\\')[1];
```

Comment:
@davenjeru why are we splitting instead of adding the path to the defined path directly?

Reply:
The layer, if it was defined using `route.use()` or if it was given an `express.Router` as its handler, will not have a path but a regex that will define the path. We split to get an array of strings and the path will be defined on the second element. Although I haven't tested it yet with paths that need params like `:id`.